btk/draw_blends.py

```diff
@@ -176,23 +179,17 @@ def __init__(
         self.batch_size = self.blend_generator.batch_size
         self.max_number = self.blend_generator.max_number
 
-        if isinstance(surveys, Survey):
-            self.surveys = [surveys]
-            self.check_compatibility(surveys)
-        elif isinstance(surveys, Iterable):
-            for s in surveys:
-                if not isinstance(s, Survey):
-                    raise TypeError(
-                        f"surveys must be a Survey object or an Iterable of Survey objects, but "
-                        f"Iterable contained object of type {type(s)}"
-                    )
-                self.check_compatibility(s)
-            self.surveys = surveys
-        else:
-            raise TypeError(
-                f"surveys must be a Survey object or an Iterable of Survey objects,"
-                f"but surveys is type {type(surveys)}"
-            )
+        # get surveys from galcheat.
+        if not isinstance(surveys, Iterable):
+            surveys = [surveys]
+
+        self.surveys = []
+        for s in surveys:
+            if not isinstance(s, str):
```

Comment:
If the PSF is now independent on the survey, shouldn't we still support giving galcheat surveys directly for experienced users ? This should not add too much complexity, I believe.

Reply:
sure I can change that

Reply:
done !